src/searchModal.js

```diff
@@ -0,0 +1,203 @@
+import infoMessageTpl from 'ui/searchModal/tpl/info-message';
+import 'ui/searchModal/css/searchModal.css';
+import component from 'ui/component';
+import 'ui/modal';
+import 'ui/datatable';
+
+/**
+ * Creates a search modal instance
+ * @param {object} config - search modal configuration
+ * @param {string} config.query - search query to be set and triggered on component creation
+ * @param {string} config.url - search endpoint
+ * @param {object} config.events - events hub
+ */
+export default function searchModalFactory(config) {
+    const instance = component().setTemplate(layoutTpl);
+    let searchInput = null;
+    let searchButton = null;
+    let clearButton = null;
+    let running = false;
+    instance.on('render', () => renderModal());
+    instance.on('destroy', () => destroyModal());
+
+    /**
+     * Creates search modal, inits template selectors, and triggers initial search
+     */
+    function renderModal() {
+        initModal();
+        initUiSelectors();
+        searchButton.trigger('click');
+    }
+
+    /**
+     * Removes search modal
+     */
+    function destroyModal() {
+        instance.getElement().removeClass('modal').modal('destroy');
+        $('.modal-bg').remove();
+    }
+
+    /**
+     * Creates search modal
+     */
+    function initModal() {
+        instance
+            .getElement()
+            .addClass('modal')
+            .on('closed.modal', function () {
+                instance.destroy();
+            })
+            .modal({
+                disableEscape: true,
+                width: $(window).width(),
+                minHeight: $(window).height(),
+                modalCloseClass: 'modal-close-left'
+            })
+            .focus();
+    }
+
+    /**
+     * Inits template selectors and sets initial search query on search input
+     */
+    function initUiSelectors() {
+        searchButton = $('.btn-search', instance.getElement());
+        clearButton = $('.btn-clear', instance.getElement());
+        searchInput = $('.search-bar-container input', instance.getElement());
+        searchButton.on('click', search);
+        clearButton.on('click', clear);
+        searchInput.val(config.query);
+    }
+
+    /**
+     * Request search results and manages its results
+     */
+    function search() {
+        const query = searchInput.val();
+        if (query === '') {
+            clear();
+            return;
+        }
+        //throttle and control to prevent sending too many requests
+        const searchHandler = _.throttle(function searchHandlerThrottled(query) {
+            if (running === false) {
+                running = true;
+                $.ajax({
+                    url: config.url,
+                    type: 'POST',
+                    data: { query: query },
+                    dataType: 'json'
+                })
+                    .done(buildSearchResultsDatatable)
+                    .complete(function () {
```

Comment:
`complete()` will be deprecated, please use `always()` instead
```suggestion
                    .always(function () {
```